object-utils.go

```diff
@@ -73,27 +73,32 @@ func IsValidBucketName(bucket string) bool {
 // - Caret ("^")
 // - Grave accent / back tick ("`")
 // - Vertical bar / pipe ("|")
+//
```

Comment:
> now, IsValidObjectName is actually stricter and isValidObjectPrefix is simpler
> IsValidObjectName = isValidObjectPrefix + restrictions (i.e trailing "/" and length 0 restrictions)

Stricter means 'strings.Contains' moved to isValidOjectPrefix, its not the right place for that check.